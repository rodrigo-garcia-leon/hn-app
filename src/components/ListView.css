.ListView ul {
  list-style-type: none;
}
