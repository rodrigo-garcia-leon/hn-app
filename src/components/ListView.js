import React, { useState, useEffect } from "react";
import List from "./List";
import { fetchList } from "../services/data";

export const LIST_TOP = "top";
const SLICE_START = 0;
const SLICE_END = 30;

function ListView() {
  const [items, setItems] = useState(null);

  useEffect(() => {
    async function loadData() {
      try {
        const items = await fetchList(LIST_TOP);
        const slice = items.slice(SLICE_START, SLICE_END);

        setItems(slice);
      } catch (error) {
        console.error(error);
      }
    }

    loadData();
  }, []);

  return <List items={items} />;
}

export default ListView;
