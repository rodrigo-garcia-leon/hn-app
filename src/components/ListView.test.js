import React from "react";
import { create, act } from "react-test-renderer";
import ListView from "./ListView";
import {
  TEST_LIST,
  TEST_ITEM_22069310,
  TEST_ITEM_22089166,
  TEST_ITEM_22089546,
  setupFetchErrorMock
} from "../services/data.test";

describe("ListView", () => {
  let element, mockFetch;

  const createElement = async () =>
    act(async () => {
      element = create(<ListView />).toJSON();
    });

  beforeEach(() => {
    mockFetch = null;
  });

  afterEach(() => {
    if (mockFetch) {
      mockFetch.mockRestore();
    }
  });

  test("empty", () => {
    element = create(<ListView />).toJSON();

    expect(element).toMatchSnapshot();
  });

  test("network error", async () => {
    mockFetch = setupFetchErrorMock();

    await createElement();
    expect(element).toMatchSnapshot();
  });

  test("ok", async () => {
    mockFetch = jest
      .spyOn(global, "fetch")
      .mockResolvedValueOnce({ json: jest.fn().mockResolvedValue(TEST_LIST) })
      .mockResolvedValueOnce({
        json: jest.fn().mockResolvedValue(TEST_ITEM_22069310)
      })
      .mockResolvedValueOnce({
        json: jest.fn().mockResolvedValue(TEST_ITEM_22089166)
      })
      .mockResolvedValueOnce({
        json: jest.fn().mockResolvedValue(TEST_ITEM_22089546)
      });

    await createElement();
    expect(element).toMatchSnapshot();
  });
});
