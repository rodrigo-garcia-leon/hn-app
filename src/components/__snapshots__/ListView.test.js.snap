// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListView empty 1`] = `null`;

exports[`ListView network error 1`] = `null`;

exports[`ListView ok 1`] = `null`;
